content/token22-in-the-client.md

```diff
@@ -0,0 +1,470 @@
+
+```ts
+const allOwnedTokens = []
+const tokenAccounts = await connection.getTokenAccountsByOwner(
+	wallet.publicKey,
+	{programId: TOKEN_PROGRAM_ID}
+)
+const tokenExtensionAccounts = await connection.getTokenAccountsByOwner(
+	wallet.publicKey,
+	{programId: TOKEN_2022_PROGRAM_ID}
+)
+
+allOwnedTokens.push(...tokenAccounts, ...tokenExtensionAccounts)
+```
+
+NOTE: It may be advised to store and associate the token program with the token upon fetching.
+
+### Check owning program
+
+You may run into a scenario where you don't know the token program for a given account. Fortunately, `getParsedAccountInfo` will allow us to determine the owning program:
+
+```tsx
+const accountInfo = await connection.getParsedAccountInfo(mintAddress);
+if (accountInfo.value === null) {
+  throw new Error('Account not found');
+}
+
+const programId = accountInfo.value.owner; // will return TOKEN_PROGRAM_ID for Token Program mint address and TOKEN_2022_PROGRAM_ID for Token Extension Program mint address
+
+//we now use the programId to fetch the tokens
+const tokenAccounts = await connection.getTokenAccountsByOwner(
+  wallet.publicKey,
+  {programId}
+)
+```
+
+NOTE: After you fetch the owning account, it may be a good idea to save that owner and associate it with the mints/tokens you are handling.
+
+# Lab - Add Extension Token support to a script
+
+Let's work through a holistic example where we add Token Extension support to an existing script. This lab will lead us through the necessary adjustments and expansions to embrace the capabilities and nuances of both the original Token Program and its extension counterpart. 
+
+By the end of this lab, we'll have navigated the complexities of supporting these two distinct but related token systems, ensuring our script can interact smoothly with both.
+
+### 1. Clone the starter code
+
+To get started, clone [this lab's repository](https://github.com/Unboxed-Software/solana-lab-token22-in-the-client/) and checkout the `starter` branch. This branch contains a couple of helper files and some boilerplate code to get you started.
+
+```bash
+git clone https://github.com/Unboxed-Software/solana-lab-token22-in-the-client.git
+cd solana-lab-token22-in-the-client
+git checkout starter
+```
+
+Run `npm install` to install the dependencies.
+
+### 2. Get familiar with the starter code
+
+The starter code comes with the following files:
+
+- `print-helpers.ts`
+- `index.ts`
+
+The **`print-helpers.ts`** file contains a function called **`printTableData`**, which is designed to output data to the console in a structured format. This function is capable of accepting any object as its argument, and it utilizes the **`console.table`** method, a feature provided by NodeJS, to display the data in an easily readable table format.
+
+Lastly, `index.ts` contains our main script. It currently only creates a connection and calls `initializeKeypair` to generate the keypair for `payer`.
```

Comment:
Use `esrun`, it's easier, see https://github.com/Unboxed-Software/solana-course/blob/main/CONTRIBUTING.md